platform.md

```diff
@@ -7,64 +7,114 @@ A platform orchestrates a lifecycle to make buildpack functionality available to
-  - [Exported Image](#exported-image)
   - [Security Considerations](#security-considerations)
   - [Additional Guidance](#additional-guidance)
     - [Environment](#environment)
-    - [Run Image Rebasing](#run-image-rebasing)
+      - [Buildpack Environment](#buildpack-environment)
+        - [Stack-Provided Variables](#stack-provided-variables)
+        - [User-Provided Variables](#user-provided-variables)
+      - [Launch Environment](#launch-environment)
     - [Caching](#caching)
   - [Data Format](#data-format)
-    - [order.toml (TOML)](#ordertoml-toml)
-    - [group.toml (TOML)](#grouptoml-toml)
-    - [project-metadata.toml (TOML)](#project-metadatatoml-toml)
+    - [Files](#files)
+      - [`analyzed.toml` (TOML)](#analyzedtoml-toml)
+      - [`group.toml` (TOML)](#grouptoml-toml)
+      - [`metadata.toml` (TOML)](#metadatatoml-toml)
+      - [`order.toml` (TOML)](#ordertoml-toml)
+      - [`plan.toml` (TOML)](#plantoml-toml)
+      - [`project-metadata.toml` (TOML)](#project-metadatatoml-toml)
+      - [`stack.toml` (TOML)](#stacktoml-toml)
+    - [Labels](#labels)
+      - [`io.buildpacks.build.metadata` (JSON)](#iobuildpacksbuildmetadata-json)
+      - [`io.buildpacks.lifecycle.metadata` (JSON)](#iobuildpackslifecyclemetadata-json)
+      - [`io.buildpacks.project.metadata` (JSON)](#iobuildpacksprojectmetadata-json)
+
 
 ## Platform API Version
 
-The Platform API version:
+This document specifies Platform API version `0.3`
+
+Platform API versions:
  - MUST be in form `<major>.<minor>` or `<major>`, where `<major>` is equivalent to `<major>.0`
- - MUST describe the implemented Platform API.
- - SHALL indicate compatibility with a given lifecycle according to the following rules:
-    - When `<major>` is `0`, the platform is only compatible with lifecycles implementing that exact Platform API.
-    - When `<major>` is greater than `0`, the platforms is only compatible with lifecycles implementing platform API
-    `<major>.<minor>`, where `<major>` of the lifecycle equals `<major>` of the platform and `<minor>` of the lifecycle
-    is greater than or equal to `<minor>` of the platform.
+ - When `<major>` is greater than `0` increments to `<minor>` SHALL exclusively indicate additive changes
 
-### Compatibility Verification
+## Terminology
 
-The lifecycle SHALL verify compatibility if the environment variable `CNB_PLATFORM_API` is set. The value of this
-environment variable MUST be the version of the Platform API the platform implements. Compatibility verification SHALL
-NOT occur if this environment variable is not set. Compatibility verification SHALL occur before any other validation.
+#### CNB Terminology
+A **buildpack** refers to software compliant with the [Buildpack Interface Specification](https://github.com/buildpacks/spec/blob/main/buildpack.md)
 
-## Stacks
+A **stack** is a contract defined by a **build image** and a **run image** that are only updated with security patches.
```

Comment:
Separate note: the contract isn't defined by the images, but by the functionality that the images provide.

Reply:
Maybe if we don't define `stack` by itself? Something like:
```suggestion
A **stack contract** is defined by a set of unchanging ABIs, command line interfaces, and functional behaviors. Images marked with a **stack ID** MUST implement the same contract.
```

Reply:
Didn't realize this line was moved and not changed. We shouldn't block merging on fixing this.

Reply:
I went to add a version of this suggestion with an accompanying definition of **stack ID** but I wasn't convinced it was easier to understand.

Given that we already have this definition in the existing spec maybe we can punt this improvement (and a possible reordering of terminology) to a future PR. It might be easier to think about these types of incremental improvements after we get this monster merged.

Reply:
Should we create an issue to track this work?